src/SURFnet/OATHBundle/Controller/OathController.php

```diff
@@ -30,7 +30,12 @@ public function getOcraChallengeAction()
             $data = $this->ocra->generateChallenge();
         } catch (\Exception $e) {
             $data = array('error' => $e->getMessage());
-            $responseCode = $e->getCode() ?: 500;
+            // only pass error code is it's a valid http status code
+            if ($e->getCode() >= 200 && $e->getCode() <= 510) {
+                $responseCode = $e->getCode() ?: 500;
+            } else {
+                $responseCode = 500;
+            };
```

Comment:
I don't say this often, but maybe it's a good idea to move this repeating block of logic to the BaseController. Adding a `getErrorCode` method there which takes the exception as a parameter.

The logic in this block seems sound. One complaint could be the use of the `else` clause. That should be avoided if not strictly necessary. In this case you could return 500 as a default and use a singular if statement to test if the error code is in the suitable HTTP range.  Leaving you with somehting like this proto code:

```
responseCode = 500
if (exceptionCode > 200 and exceptionCode <= 510) {
   responseCode = exceptionCode // no need here for the ?: 500 construction. That was already verified in the 'parent' condition
}
return responseCode